src/ksm/client/client_test.go

```diff
@@ -60,3 +84,27 @@ func testHTTPServer(t *testing.T, requestsReceived *int, timeout time.Duration)
 
 	return testServer
 }
+
+func testKsmEndpoint(t *testing.T) *httptest.Server {
+	t.Helper()
+
+	testServer := httptest.NewServer(http.HandlerFunc(
+		func(w http.ResponseWriter, r *http.Request) {
+			w.WriteHeader(http.StatusOK)
+			_, err := io.WriteString(w,
+				`# HELP kube_pod_status_phase The pods current phase. 
+				 # TYPE kube_pod_status_phase gauge
+				 kube_pod_status_phase{namespace="default",pod="123456789"} 1
+				 # HELP kube_custom_elasticsearch_health_status Elasticsearch CRD health status
+				 # TYPE kube_custom_elasticsearch_health_status stateset
+				 kube_custom_elasticsearch_health_status {customresource_group="elasticsearch.k8s.elastic.co"} 1
+				`)
```

Comment:
can you add a comment above this function to clarify what the expected outcome here is?
and comment the logged error in the PR description to clarify that the parsing passed for gauge and failed for stateset?

Reply:
Will do